scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2140,65 @@ def visit_classdef(self, node):
             self.add_message('no-dunder-metaclass', node=node)
 
 
+class DisallowHandlerWithoutSchema(checkers.BaseChecker):
+    """Custom pylint checker prohibiting handlers which do not have schema
+    defined within the class.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'disallow-handlers-without-schema'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'Please add schema in URL_ARGS_PATH_SCHEMA for %s class.',
+            'no-schema-for-url-path-elements',
+            'Please add schema for url path elements by folllowing wiki link- '
+            'https://github.com/oppia/oppia/wiki/Validation-of-handler-args'
+        ),
+        'C0036': (
+            'Please add schema in HANDLER_ARGS_SCHEMAS for %s class.',
+            'no-schema-for-handler-args',
+            'Please add schema for payload arguments and url query string '
+            'parameters by following the link- '
+            'https://github.com/oppia/oppia/wiki/Validation-of-handler-args'
+        )
+    }
+    handler_class_names_with_no_schema = (
+        payload_validator.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA)
+
+    def check_parent_class_is_basehandler(self, node):
+        """Checks whether a class is child class of BaseHandler."""
+        return 'base.BaseHandler' in node.basenames
+
+    def visit_classdef(self, node):
+        """Visit each class definition in controllers layer module and check
+        if it contains schema or not.
+
+        Args:
+            node: astroid.nodes.ClassDef. Node for a class definition
+                in the AST.
+        """
+        basehandler_child_class = self.check_parent_class_is_basehandler(node)
+        if basehandler_child_class is False:
+            for j in node.ancestors():
+                if self.check_parent_class_is_basehandler(j):
+                    basehandler_child_class = True
+                    break
+        if basehandler_child_class is False:
+            return
+
+        if node.name in self.handler_class_names_with_no_schema:
+            return
+
+        if 'URL_PATH_ARGS_SCHEMAS' not in node.locals:
```

Comment:
Don't we also need to verify its type and value too? What I mean by that is:
1. Should `URL_PATH_ARGS_SCHEMAS` always be a dict? Should the linter fail if someone sets it to be an integer or something else?
2. Is the case where this dict is set to `{}` valid?

If the answer is Yes to any of these questions then this check is not enough and you might have to explore more to increase the check strictness.

Reply:
The functionality of SVS architecture is heavily dependent on backend tests, like the entire behaviour in context of schema validation is testedby backend tests, linters is provided to check whether any handler have had schemas defined if it is not in the list of HANDLER_CLASS_NAMES_WITH_NO_SCHEMA.
Thus the tests are written accordingly.
Looking forward for your point.

Reply:
I am fine with excluding point 2, but still concerned about point 1. What would happen if someone defines:
```python
URL_PATH_ARGS_SCHEMAS = []
```
Will backend tests fail for this case? I just want to know whether or not we can force the user to define only a dictionary with the name `URL_PATH_ARGS_SCHEMAS` and not let them getaway by defining a list or something else with the same name.

(cc @DubeySandeep what do you think about point (1) and (2) - should we handle these under linters?)

Reply:
I have added linters to verify URL_PATH_ARGS_SCHEMA and HANDLER_ARGS_SCHEMAS must be of type dict, please take a look.

Reply:
The code you have written looks correct but is written in a very complicated manner.
Why not have a common method `check_given_variable_is_a_dict(node, variable_name)` - where you pass the node and variable name e.g. URL_PATH_ARGS_SCHEMA and check if it's a dict or not. This will avoid code duplication.

Reply:
Thanks for your great suggestion, I have updated the method. PLease take a look.